index/writer.go

```diff
@@ -77,20 +78,22 @@ type indexWriter struct {
 // provide keys in increasing lexicographic order.  Duplicate keys are not
 // allowed but will not be detected.  Close() or Abort() must be called when
 // done writing.
-func NewWriter(zctx *zed.Context, engine storage.Engine, path string, options ...Option) (*Writer, error) {
-	return NewWriterWithContext(context.Background(), zctx, engine, path, options...)
+func NewWriter(zctx *zed.Context, engine storage.Engine, path string, keys field.List, options ...Option) (*Writer, error) {
+	return NewWriterWithContext(context.Background(), zctx, engine, path, keys, options...)
 }
 
-func NewWriterWithContext(ctx context.Context, zctx *zed.Context, engine storage.Engine, path string, options ...Option) (*Writer, error) {
+func NewWriterWithContext(ctx context.Context, zctx *zed.Context, engine storage.Engine, path string, keys field.List, options ...Option) (*Writer, error) {
 	w := &Writer{
-		zctx:   zctx,
-		engine: engine,
+		zctx:       zctx,
+		childField: uniqChildField(keys),
+		engine:     engine,
+		keys:       keys,
```

Comment:
Nit: Follow field declaration order.
```suggestion
		keys:       keys,
		zctx:       zctx,
		engine:     engine,
		childField: uniqChildField(keys),
```